### PR TITLE
Update code to php7

### PR DESCRIPTION
**Pull request by jonasdekeukelaere**
First part (other part wil be in Framework itself) of updating code to PHP 7.1

Will fixed https://github.com/sumocoders/FrameworkMultiUserBundle/issues/66

### CODE REVIEW

File: Controller/UserController.php [hunk 1]
```diff
@@ -77,13 +67,13 @@ public function __construct(
...
-     * @param int $id
+     * @param int|null $id
      *
      * @Template()
      *
-     * @return array
+     * @return array|RedirectResponse
      */
-    public function baseAction(Request $request, $id = null)
+    public function baseAction(Request $request, ?int $id = null)
```
[the remark this answers is not in the document]
Comment (jonasdekeukelaere) [reply Controller/UserController.php]: I don't know, I think it's the valid php7 notation

[the remark this answers is not in the document]
Comment (bjorvack) [reply Controller/UserController.php]: When using `?` you can't ommit that parameter, your required to add `null`. So I think removing the `?` would be better in this scenario

[the remark this answers is not in the document]
Comment (jonasdekeukelaere) [reply Controller/UserController.php]: that's why the "= null" is still there, but it seems just weird to have the "?" there sometimes, and other times not..

File: Controller/UserController.php [hunk 2]
```diff
@@ -63,7 +53,7 @@ public function __construct(
         string $form,
         Handler $handler,
         UserRepository $userRepository,
-        $redirectRoute = null
+        ?string $redirectRoute = null
```
[the remark this answers is not in the document]
Comment (jonasdekeukelaere) [reply Controller/UserController.php]: Fixed 5477bfc

File: Controller/UserController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasdekeukelaere) [reply Controller/UserController.php]: Fixed 5477bfc

File: Entity/BaseUser.php [hunk 3]
```diff
@@ -88,12 +88,12 @@ class BaseUser implements User
...
-        $plainPassword,
-        $displayName,
-        $email,
-        $id = null,
-        PasswordResetToken $token = null
+        string $username,
+        string $plainPassword,
+        string $displayName,
+        string $email,
+        ?int $id = null,
```
[the remark this answers is not in the document]
Comment (jonasdekeukelaere) [reply Entity/BaseUser.php]: Fixed 5477bfc

**Comment by StijnVrolijk** [comment]
@jonasdekeukelaere Fix je juist de tests nog door Travis te laten testen op PHP 7.1 en niet PHP 7.0?

**Comment by jonasdekeukelaere** [comment]
@StijnVrolijk PHP version updated to 7.1 and fixed the tests (last 7 commits)

**Pull request merged**